components/equihash/src/tromp.rs

```diff
@@ -125,6 +295,17 @@ mod tests {
             println!("Found {} solutions:", solutions.len());
             for solution in solutions {
                 println!("- {:?}", solution);
+                crate::is_valid_solution(200, 9, input, &nonce, &solution).unwrap_or_else(
+                    |error| {
+                        panic!(
+                            "unexpected invalid equihash 200, 9 solution:\n\
+                             error: {error:?}\n\
+                             input: {input:?}\n\
+                             nonce: {nonce:?}\n\
+                             solution: {solution:?}"
+                        )
+                    },
+                );
```

Comment:
I haven't got this far yet in the tests, but if the code I've added works, we will be able to verify that the generated solutions are valid.